community_usecase/Profile_generation/run_profile_generation.py

```diff
@@ -0,0 +1,394 @@
+        description="Links to academic and professional presence (e.g., "
+                    "Google Scholar, ResearchGate, LinkedIn)"
+    )
+
+
+class ScholarProfile(BaseModel):
+    personal_information: PersonalInformation
+    biography: Biography
+    research_interests: ResearchInterests
+    awards_and_distinctions: AwardsAndDistinctions
+    education: Education
+    links: Links
+
+
+def generate_html_profile(input_text,
+                          template_path: str = "template.html",
+                          output_file: str = "profile.html",
+                          base_url: str = "https://cemse.kaust.edu.sa") -> \
+        None:
+    """Fill an HTML template with profile data, some with agent rewriting,
+    and write to file."""
+    extraction_prompt = (
+        "You are a scholarly‑profile extraction assistant. "
+        "Return ONLY JSON that matches the provided schema."
+    )
+
+    extractor = ChatAgent(system_message=extraction_prompt)
+    extraction_response = extractor.step(
+        input_message=input_text,
+        response_format=ScholarProfile,
+    )
+    profile = extraction_response.msgs[0].parsed
+
+    def to_html_list(items: list[str]) -> str:
+        return "<br>".join(items)
+
+    def format_awards_with_agent(awards: list[str]) -> str:
+        formatted_awards = []
+        agent = ChatAgent(
+            system_message="""Reformat awards as HTML list items using the 
+                           structure, Like:
+                           <li class="field__item"><strong>The 
+                           NSF CAREER grant in low-power computing and 
+                           communication systems</strong>, 2024</li>.""")
+        for award in awards:
+            response = agent.step(f"Reformat this: {award}")
+            formatted_awards.append(response.msgs[0].content.strip())
+        return "\n".join(formatted_awards)
+
+    def format_education_with_agent(degrees: list[str]) -> str:
+        agent = ChatAgent(
+            system_message="""Reformat education degrees into <dl> format. 
+            Like: 
+            <dl class="field__item"><dt>Doctor of Philosophy (Ph.D.)</dt>
+            <dd class="text-break">Integrated Circuits and Systems, 
+            <em>University of California</em>, United States, 2003</dd></dl>
+            """)
+
+        items = []
+        for degree in degrees:
+            response = agent.step(f"Reformat this: {degree}")
+            items.append(response.msgs[0].content.strip())
+        return "\n".join(items)
+
+    def format_links(links: list[str], html_template: str) -> str:
+        agent = ChatAgent(
+            system_message="Reformat link into this html structure: %s ." %
+                           html_template)
+        items = []
+        for link in links:
+            response = agent.step(f"Reformat this: {link}")
+            items.append(response.msgs[0].content.strip())
+        return "\n".join(items)
```

Comment:
Do not do it in multiple steps. Use response format to format the awards in one shot. Otherwise, it is too exspensive